rct229/rulesets/ashrae9012019/ruleset_functions/get_hvac_systems_primarily_serving_comp_room.py

```diff
@@ -1,70 +1,145 @@
-
-    Parameters
-    ----------
-    rmd dict
-        A dictionary representing a ruleset model description as defined by the ASHRAE229 schema
-
-    Returns
-    -------
-    hvac_systems_primarily_serving_comp_rooms_list list
-        A list of hvac systems in which greater than 50% of the area served by the HVAC system is computer room space.
+    Returns a list of HVAC systems in which the computer room space loads are the dominant loads for an HVAC system
+    serving multiple spaces including computer rooms and non computer rooms (i.e., computer room spaces account for greater than 50% of cooling load).
+
+     Parameters
+     ----------
+     rmd dict
+         A dictionary representing a ruleset model description as defined by the ASHRAE229 schema
+
+     Returns
+     -------
+     hvac_systems_primarily_serving_comp_rooms_list list
+         A list of hvac systems in which the computer room space loads are the dominant loads for an HVAC system serving multiple spaces
+         including computer rooms and non computer rooms (i.e., computer room spaces account for greater than 50% of cooling load).
     """
+
     hvac_zone_list_w_area_dict = get_hvac_zone_list_w_area_by_rmd_dict(rmd)
-    hvac_systems_primarily_serving_comp_rooms_list = []
+    zone_with_computer_room_list = list(
+        {
+            zone["id"]
+            for zone in find_all("$.buildings[*].building_segments[*].zones[*]", rmd)
+            for space in find_all("$.spaces[*]", zone)
+            if space.get("lighting_space_type") == LIGHTING_SPACE_OPTION.COMPUTER_ROOM
+        }
+    )
 
-    for hvac in find_all(
-        "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+    hvac_systems_primarily_serving_comp_rooms_list = []
+    for hvac_id in find_all(
+        "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
         rmd,
     ):
         assert_(
-            hvac_zone_list_w_area_dict.get(hvac["id"]),
-            f"HVAC system {hvac['id']} is missing in the  zone.terminals data group.",
-        )
-        hvac_sys_total_floor_area = hvac_zone_list_w_area_dict[hvac["id"]]["total_area"]
-        hvac_sys_zone_id_list = hvac_zone_list_w_area_dict[hvac["id"]]["zone_list"]
-
-        hvac_zones_list = [
-            find_exactly_one_zone(rmd, zone_id) for zone_id in hvac_sys_zone_id_list
-        ]
-        hvac_spaces_list = [
-            space for zone in hvac_zones_list for space in find_all("$.spaces[*]", zone)
-        ]
-
-        hvac_system_computer_room_floor_area = sum(
-            [
-                space.get("floor_area", ZERO.AREA)
-                for space in hvac_spaces_list
-                if space.get("lighting_space_type")
-                == LightingSpaceOptionsG37.COMPUTER_ROOM
-            ],
-            ZERO.AREA,
+            hvac_zone_list_w_area_dict.get(hvac_id),
+            f"HVAC system {hvac_id} is missing in the zone.terminals data group.",
```

Comment:
Is this message correct?
Maybe change to `HVAC system {hvac_id} is not assigned to any thermal zones. Check zone.terminals data group.`